app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
That's how HTML works. URL parameters aren't meant to be super user friendly, either. It's perfectly fine to have parameters with empty values in there.